code/modules/shuttle/custom_shuttle/custom_shuttle.dm

```diff
@@ -0,0 +1,58 @@
+/obj/machinery/computer/shuttle/custom_shuttle
+	name = "Nanotrasen Flight Control System"
+	desc = "A shuttle control computer."
+	icon_screen = "shuttle"
+	icon_keyboard = "tech_key"
+	shuttleId = ""
+	light_color = LIGHT_COLOR_CYAN
+	req_access = list()
+	interaction_flags_machine = INTERACT_MACHINE_ALLOW_SILICON
+
+
+/obj/machinery/computer/shuttle/custom_shuttle/proc/linkShuttle(new_id)
+	if(shuttleId=="")
+		shuttleId = new_id
+		possible_destinations = "whiteship_home;shuttle[new_id]_custom;"
+		return TRUE
+	return FALSE
+
+
+//docking cam
+/obj/machinery/computer/camera_advanced/shuttle_docker/custom
+	name = "Shuttle Navigation Computer"
+	desc = "Used to designate a precise transit location for private ships."
+	lock_override = NONE
+	whitelist_turfs = list(/turf/open/space, /turf/open/lava)
+	jump_to_ports = list("whiteship_home" = 1)
+	view_range = 12
+	designate_time = 30
+	circuit = /obj/item/circuitboard/computer/shuttle/docker
+
+/obj/machinery/computer/camera_advanced/shuttle_docker/custom/Initialize(mapload)
+	. = ..()
+	GLOB.jam_on_wardec += src
```

Comment:
This is nice and all, but it only works if the computer was made before the nuclear war declaration, yet we don't have a global variable for that so you have to add it.